cf-net/cf-net.c

```diff
@@ -0,0 +1,507 @@
+        };
+        int err = 0;
+        char *buf = xstrdup(server);
+        char *host, *port;
+        ParseHostPort(buf, &host, &port);
+        if (port == NULL)
+        {
+            port = CFENGINE_PORT_STR;
+        }
+        conn = ServerConnection(host, port, 30, connflags, &err);
+        free(buf);
+    }
+    if (conn == NULL)
+    {
+        nprint("Failed to connect to '%s'.\n", server);
+        return NULL;
+    }
+    return conn;
+}
+
+int CFNetConnectSingle(char *server)
+{
+    AgentConnection *conn = CFNetOpenConnection(server);
+    if(conn == NULL)
+    {
+        return -1;
+    }
+    nprint("Connected & authenticated successfully to '%s'.\n", server);
+    DeleteAgentConn(conn);
+    return 0;
+}
+
+int CFNetConnect(char **args)
+{
+    for (int i = 0; args[i] != NULL; ++i)
+    {
+        cfnet_silent = false;
+        CFNetConnectSingle(args[i]);
+        cfnet_silent = true;
+    }
+    return 0;
+}
+
+void CFNetStatPrint(const char* file, int st_mode)
+{
+    if (S_ISDIR(st_mode))
+    {
+        nprint("'%s' is a directory.\n", file);
+    }
+    else if (S_ISREG(st_mode))
+    {
+        nprint("'%s' is a regular file.\n", file);
+    }
+    else if (S_ISSOCK(st_mode))
+    {
+        nprint("'%s' is a socket.\n", file);
+    }
+    else if (S_ISCHR(st_mode))
+    {
+        nprint("'%s' is a character device file.\n", file);
+    }
+    else if (S_ISBLK(st_mode))
+    {
+        nprint("'%s' is a block device file.\n", file);
+    }
+    else if (S_ISFIFO(st_mode))
+    {
+        nprint("'%s' is a named pipe (FIFO).\n", file);
+    }
+    else if (S_ISLNK(st_mode))
+    {
+        nprint("'%s' is a symoblic link.\n", file);
+    }
+    else
+    {
+        nprint("'%s' has an unrecognized st_mode.\n", file);
+    }
+}
+
+int CFNetStat(char **args)
+{
+    char *server = args[0];
+    char *file = args[1];
+    AgentConnection *conn = CFNetOpenConnection(server);
+    if (conn == NULL)
+    {
+        return -1;
+    }
+    bool encrypt = true;
+    struct stat sb;
+    int r = cf_remote_stat(conn, encrypt, file, &sb, "file");
+    if ( r!= 0 )
+    {
+        nprint("Could not stat: '%s'.\n", file);
+        // nprint("(Check cf-serverd output for more info)\n");
+    }
+    else
+    {
+        Log(LOG_LEVEL_INFO, "Detailed stat output:\n"
+                            "mode  = %jo, \tsize = %jd,\n"
+                            "uid   = %ju, \tgid = %ju,\n"
+                            "atime = %jd, \tmtime = %jd\n",
+            (uintmax_t)sb.st_mode,(intmax_t)sb.st_size,
+            (uintmax_t)sb.st_uid, (uintmax_t)sb.st_gid,
+            (intmax_t)sb.st_atime,(intmax_t) sb.st_mtime);
+        CFNetStatPrint(file, sb.st_mode);
+    }
+    DeleteAgentConn(conn);
+    return 0;
+}
+
+int JustConnect(char *server, char* port)
+{
+    char txtaddr[CF_MAX_IP_LEN] = "";
+    return SocketConnect(server, port, 100, true,
+                         txtaddr, sizeof(txtaddr));
+}
+
+int CFNetMulti(char *server)
```

Comment:
I like this new command, as it already found a bug! However I don't think it should be a separate "multi" command. It could be a `--continuous` or `--multi` option, since it can be applied to most of the commands, potentially finding more bugs. We can discuss how it would apply to other commands. For the time being, leave it even as a separate command doing only simple connects, since it's very useful.